### PR TITLE
Donations bug scrub - Sanitize and escape plugin

### PR DESCRIPTION
**Pull request by MaximilianoRicoTabo**
* Make min price and max price number fields.
 * Sanitize some requests around

### All Submissions:

* [x] Have you followed the [Contributing guidelines](https://github.com/strangerstudios/pmpro-pay-by-check/blob/dev/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/pmpro-pay-by-check/pulls/) for the same update/change?

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: includes/level-settings.php [hunk 1]
```diff
@@ -32,13 +32,13 @@ function pmprodon_pmpro_membership_level_after_other_settings() {
 	<tr>
 		<th scope="row" valign="top"><label for="donation_min_price"><?php _e( 'Min Amount:', 'pmpro-donations' ); ?></label></th>
 		<td>
-			<?php echo $pmpro_currency_symbol; ?><input type="text" id="donation_min_price" name="donation_min_price" value="<?php echo esc_attr( $min_price ); ?>" />
+			<?php echo $pmpro_currency_symbol; ?><input type="number" id="donation_min_price" name="donation_min_price" value="<?php echo esc_attr( $min_price ); ?>" />
```
Comment (andrewlimaza) [inline includes/level-settings.php:35]: For these changes, I think we have to stick to text so people can add float values. The preg_replace for these values sanitize it already as it only allows numbers and periods.

File: includes/checkout.php [hunk 2]
```diff
@@ -299,7 +299,7 @@ function pmprodon_unhook_pmpro_level_cost_text() {
  */
 function pmprodon_pmpro_checkout_order( $order ) {
 	if ( ! empty( $_REQUEST['donation'] ) ) {
-		$donation = preg_replace( '/[^0-9\.]/', '', $_REQUEST['donation'] );
+		$donation = sanitize_text_field( preg_replace( '/[^0-9\.]/', '', $_REQUEST['donation'] ) );
```
Comment (andrewlimaza) [inline includes/checkout.php:302]: The preg_replace should be enough here to keep the data sanitized.

**Review by andrewlimaza — CHANGES_REQUESTED** [review]

File: includes/level-settings.php (hunk 1 above)
Comment (MaximilianoRicoTabo) [reply includes/level-settings.php:35]: type="number" allow users to add float values.  also text with the preg replace let users save just dots.
Also mobile keyboards for number fields are great and it's and enhancement in terms of UX and semantically seems more adequate.


image

image

Comment (dparker1005) [reply includes/level-settings.php:35]: Keeping payment amount fields as `text` fields is consistent with core PMPro:
https://github.com/strangerstudios/paid-memberships-pro/blob/fd6bf1c18315c71013da26d4ab8c0f24e6640591/adminpages/membershiplevels.php#L550

If we want to switch billing amount fields to `number` fields, this should be discussed and implemented in core PMPro first, and then we can start updating Add Ons

File: includes/level-settings.php [hunk 3]
```diff
@@ -76,10 +76,10 @@ function pmprodon_pmpro_save_membership_level( $level_id ) {
 	} else {
 		$donations_only = 0;
 	}
-	$min_price       = preg_replace( '[^0-9\.]', '', $_REQUEST['donation_min_price'] );
-	$max_price       = preg_replace( '[^0-9\.]', '', $_REQUEST['donation_max_price'] );
+	$min_price       = $_REQUEST['donation_min_price'];
+	$max_price       = $_REQUEST['donation_max_price'];
```
Comment (dparker1005) [inline includes/level-settings.php:80]: We need sanitization here. Even if these fields are `number` HTML fields, the HTML source can still be edited to make them `text` and non-numeric data could still be submitted. It is not a likely scenario, but we should still be protecting against that

**Review by dparker1005 — CHANGES_REQUESTED** [review]

**Pull request merged**